docs/source/user/parallel.rst

```diff
@@ -156,4 +156,328 @@ it would require a pervasive change that rewrites the code to extract kernel
+    identified parallel loops. From the example, ``#0`` is ``np.sin``, ``#1``
+    is ``np.cos`` and ``#2`` and ``#3`` are ``prange()``:
+
+    .. code-block:: python
+
+        Parallel loop listing for  Function test, example.py (4) 
+        --------------------------------------|loop #ID
+        @njit(parallel=True)                  | 
+        def test(x):                          | 
+            n = x.shape[0]                    | 
+            a = np.sin(x)---------------------| #0
+            b = np.cos(a * a)-----------------| #1
+            acc = 0                           | 
+            for i in prange(n - 2):-----------| #3
+                for j in prange(n - 1):-------| #2
+                    acc += b[i] + b[j + 1]    | 
+            return acc                        | 
+
+    It is worth noting that the loop IDs are enumerated in the order they are
+    discovered which is not necessarily the same order as present in the source.
+    Further, it should also be noted that the parallel transforms use a static
+    counter for loop ID indexing. As a consequence it is possible for the loop
+    ID index to not start at 0 due to use of the same counter for internal
+    optimizations/transforms taking place that are invisible to the user.
+
+#. Fusing loops
+    This section describes the attempts made at fusing discovered
+    loops noting which succeeded and which failed. In the case of failure to
+    fuse a reason is given (e.g. dependency on other data). From the example:
+
+    .. code-block:: text
+
+        --------------------------------- Fusing loops ---------------------------------
+        Attempting fusion of parallel loops (combines loops with similar properties)...
+        Trying to fuse loops #0 and #1:
+            - fusion succeeded: parallel for-loop #1 is fused into for-loop #0.
+        Trying to fuse loops #0 and #3:
+            - fusion failed: loop dimension mismatched in axis 0. slice(0, x_size0.1, 1)
+        != slice(0, $40.4, 1)
+
+    It can be seen that fusion of loops ``#0`` and ``#1`` was attempted and this
+    succeeded (both are based on the same dimensions of ``x``). Following the
+    successful fusion of ``#0`` and ``#1``, fusion was attempted between ``#0``
+    (now including the fused ``#1`` loop) and ``#3``. This fusion failed because
+    there is a loop dimension mismatch, ``#0`` is size ``x.shape`` whereas
+    ``#3`` is size ``x.shape[0] - 2``.
+
+#. Before Optimization
+    This section shows the structure of the parallel regions in the code before
+    any optimization has taken place, but with loops associated with their final
+    parallel region (this is to make before/after optimization output directly
+    comparable). Multiple parallel regions may exist if there are loops which
+    cannot be fused, in this case each region will still execute in parallel,
+    each running in the sequence presented. From the example:
```

Comment:
I understand what you mean here, but the phrasing is a little confusing.  Perhaps something more like: "in this case code within each region will execute in parallel, but each parallel region will run sequentially."

Reply:
Good idea, that's much better wording, thanks for the suggestion, patched in 23b86a021